scserversocket.js

```diff
@@ -461,6 +461,7 @@ SCServerSocket.prototype.setAuthToken = function (data, options, callback) {
       var tokenData = {
         token: signedToken
       };
+      self.signedAuthToken = signedToken
```

Comment:
This part of the code could potentially be executed asynchronously, that means that it's possible that the `socket.authToken` could have been unset (or changed) while the original token was being still being signed. So it's good to add an extra if statement to check this. Also, missing semicolon.